### PR TITLE
chore(rxSortableColumn): decompose monolithic source

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-363

### LGTMS
- [x] Dev LGTM
- [x] Dev LGTM

### CODE REVIEW

File: src/rxSortableColumn/rxSortableColumn.js [hunk 1]
```diff
@@ -1,111 +0,0 @@
...
- * @param {String} displayText - The text to be displayed in the link
- * @param {Function} sortMethod - The sort function to be called when the link is
- * clicked
- * @param {String} sortProperty - The property on the array to sort by when the
- * link is clicked.
- * @param {Object} predicate - The current property the collection is sorted by.
- * @param {Boolean} reverse - Indicates whether the collection should sort the
- * array in reverse order.
- */
-.directive('rxSortableColumn', function () {
```
Comment (CITguy) [inline src/rxSortableColumn/rxSortableColumn.js]: moved to `src/rxSortableColumn/scripts/rxSortableColumn.js`

File: src/rxSortableColumn/rxSortableColumn.js [hunk 2]
```diff
@@ -1,111 +0,0 @@
...
- * @description
- * Service which provided utility methods for sorting collections.
- *
- * @example
- * <pre>
- * rxSortUtil.getDefault() // returns a sort object with name as the default.
- * rxSortUtil.sortCol($scope, 'name') // sorts the collection based on the predicate
- * </pre>
- */
-.factory('rxSortUtil', function () {
```
Comment (CITguy) [inline src/rxSortableColumn/rxSortableColumn.js]: moved to `src/rxSortableColumn/scripts/rxSortUtil.js`

File: src/rxSortableColumn/rxSortableColumn.js [hunk 3]
```diff
@@ -1,111 +0,0 @@
...
- * @example
- * <pre>
- * [{ name: { firstName: 'Adam' } }, { }] | rxSortEmptyTop 'name.firstName':false
- * Will sort as [{}, { name: { firstName: 'Adam' } }].
- *
- * [{ name: { firstName: 'Adam' } }, { name: { firstName: null } ] | rxSortEmptyTop 'name.firstName':true
- * Will sort as [{ name: { firstName: 'Adam' } }, {}]
- * </pre>
- */
-.filter('rxSortEmptyTop', ['$filter', '$parse', function ($filter, $parse) {
```
Comment (CITguy) [inline src/rxSortableColumn/rxSortableColumn.js]: moved to `src/rxSortableColumn/scripts/rxSortEmptyTop.js`

File: src/rxSortableColumn/rxSortableColumn.spec.js [hunk 4]
```diff
@@ -1,126 +0,0 @@
-/* jshint node: true */
-
-describe('rxSortableColumn', function () {
```
Comment (CITguy) [inline src/rxSortableColumn/rxSortableColumn.spec.js]: moved to `src/rxSortableColumn/scripts/rxSortableColumn.spec.js`

File: src/rxSortableColumn/rxSortableColumn.spec.js [hunk 5]
```diff
@@ -1,126 +0,0 @@
...
-    it('should render template correctly', function () {
-        expect(el).not.be.empty;
-        expect(el.find('button')).not.be.empty;
-        expect(el.find('button').text()).to.contain('Yo!');
-        expect(el.find('i')).not.be.empty;
-        expect(el.find('i').hasClass('asc')).to.be.true;
-    });
-});
-
-describe('rxSortUtil', function () {
```
Comment (CITguy) [inline src/rxSortableColumn/rxSortableColumn.spec.js]: moved to `src/rxSortableColumn/scripts/rxSortUtil.spec.js`

File: src/rxSortableColumn/rxSortableColumn.spec.js [hunk 6]
```diff
@@ -1,126 +0,0 @@
...
-
-    it('SortUtil: sortCol will set reverse to false if it is different predicate', function () {
-        var scope = { sort: { predicate: 'name', reverse: true }, pager: { pageNumber: 1 }};
-
-        util.sortCol(scope, 'datacenter');
-        expect(scope.sort.reverse).to.be.false;
-    });
-});
-
-describe('rxSortEmptyTop', function () {
```
Comment (CITguy) [inline src/rxSortableColumn/rxSortableColumn.spec.js]: moved to `src/rxSortableColumn/scripts/rxSortEmptyTop.spec.js`

**Comment by bry4n** [comment]
LGTM

**Comment by 100stacks** [comment]
LGTM

**Pull request merged**